integrationTests/singleShard/block/executingMiniblocksSc_test.go

```diff
@@ -60,58 +71,124 @@ func TestShouldProcessWithScTxsJoinAndRewardTheOwner(t *testing.T) {
-	proposeBlock(nodes, idxProposer, round)
-	syncBlock(t, nodes, idxProposer, round)
+	checkRootHashes(t, nodes, []int{0})
+
+	time.Sleep(1 * time.Second)
+}
+
+func TestProcessesJoinGameTheSamePlayerMultipleTimesRewardAndEndgameInMultipleRounds(t *testing.T) {
+	t.Skip("this is a stress test for VM and AGAR.IO")
+
+	stepDelay = time.Nanosecond
+
+	p := profile.Start(profile.MemProfile, profile.ProfilePath("."), profile.NoShutdownHook)
+	defer p.Stop()
+
+	log := logger.DefaultLogger()
+	log.SetLevel(logger.LogDebug)
+
+	scCode, err := ioutil.ReadFile(agarioFile)
+	assert.Nil(t, err)
+
+	maxShards := uint32(1)
+	numOfNodes := 4
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+
+	nodes := make([]*integrationTests.TestProcessorNode, numOfNodes)
+	for i := 0; i < numOfNodes; i++ {
+		nodes[i] = integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+	}
+
+	idxProposer := 0
+	nrPlayers := 100
+	players := make([]*integrationTests.TestWalletAccount, nrPlayers)
+	players[0] = integrationTests.CreateTestWalletAccount(nodes[idxProposer].ShardCoordinator, 0)
+	for i := 1; i < nrPlayers; i++ {
+		players[i] = players[0]
+	}
+	nrPlayers = 1
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+	}
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(time.Second)
+
+	round := uint64(0)
+	round = incrementAndPrintRound(round)
+
+	hardCodedSk, _ := hex.DecodeString("5561d28b0d89fa425bbbf9e49a018b5d1e4a462c03d2efce60faf9ddece2af06")
+	hardCodedScResultingAddress, _ := hex.DecodeString("000000000000000000005fed9c659422cd8429ce92f8973bba2a9fb51e0eb3a1")
+	nodes[idxProposer].LoadTxSignSkBytes(hardCodedSk)
+
+	initialVal := big.NewInt(10000000)
+	topUpValue := big.NewInt(500)
+	integrationTests.MintAllNodes(nodes, initialVal)
+	integrationTests.MintAllPlayers(nodes, players, initialVal)
+
+	deployScTx(nodes, idxProposer, string(scCode))
+	time.Sleep(time.Second)
+	proposeBlock(nodes, []int{idxProposer}, round)
+	time.Sleep(time.Second)
+	syncBlock(t, nodes, []int{idxProposer}, round)
 	round = incrementAndPrintRound(round)
 
-	checkRewardIsDoneCorrectly(
+	runMultipleRoundsOfTheGame(
 		t,
+		10,
+		nrPlayers,
 		nodes,
-		idxProposer,
-		idxProposer,
-		initialVal,
+		players,
 		topUpValue,
-		withdrawValue,
 		hardCodedScResultingAddress,
+		round,
+		maxShards,
+		[]int{idxProposer},
 	)
 
 	checkRootHashes(t, nodes, []int{0})
 
-	time.Sleep(1 * time.Second)
+	time.Sleep(time.Second)
 }
 
-func TestProcessesJoinGameOf100PlayersRewardAndEndgame(t *testing.T) {
+func TestProcessesJoinGame100PlayersMultipleTimesRewardAndEndgameInMultipleRounds(t *testing.T) {
 	t.Skip("this is a stress test for VM and AGAR.IO")
 
 	stepDelay = time.Nanosecond
```

Comment:
Should not rewrite stepDelay

Reply:
stepDelay is used for every transaction dispatching, I put that to 1 nanosecond, than have a bigger stepDelay between transaction dispacthing to block proposing and to sync. Otherwise it takes 100seconds to dispatch 100 transactions. I will delete stepDelay between transaction, and use the one second general stepDelay.